packages/piro/test/MockModelEval_A_Tpetra.hpp

```diff
@@ -0,0 +1,174 @@
+// @HEADER
+// ************************************************************************
+//
+//        Piro: Strategy package for embedded analysis capabilitites
+//                  Copyright (2010) Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Andy Salinger (agsalin@sandia.gov), Sandia
+// National Laboratories.
+//
+// ************************************************************************
+// @HEADER
+
+#ifndef MOCKMODELEVAL_A_TPETRA_H
+#define MOCKMODELEVAL_A_TPETRA_H
+
+#include "Teuchos_Assert.hpp"
+#include "Teuchos_RCP.hpp"
+#include "Thyra_ModelEvaluatorDefaultBase.hpp"
+#include "Tpetra_MultiVector.hpp"
+#include "Tpetra_CrsMatrix.hpp"
+#include "Thyra_TpetraThyraWrappers.hpp"
+
+
+typedef Tpetra::Map<int,int>  Tpetra_Map;
+typedef Tpetra::Vector<double,int,int>  Tpetra_Vector;
+typedef Tpetra::MultiVector<double,int,int>  Tpetra_MultiVector;
+typedef Tpetra::Operator<double,int,int>  Tpetra_Operator;
+typedef Tpetra::CrsGraph<int,int>  Tpetra_CrsGraph;
+typedef Tpetra::CrsMatrix<double,int,int>  Tpetra_CrsMatrix;
+typedef Thyra::TpetraOperatorVectorExtraction<
+    double, int, int> ConverterT;
+
+/** \brief Epetra-based Model Evaluator subclass for Charon!
```

Comment:
Why the Comment about Charon? Should probably be removed.

Reply:
Thanks, it's a copy and paste leftover. I'll get rid of it.